src/ui/map.js

```diff
@@ -2618,16 +2622,31 @@ class Map extends Camera {
         return this.style.getFeatureState(feature);
     }
 
-    _containerDimensions() {
-        let width = 0;
-        let height = 0;
-
+    _containerDimensions(): any {
         if (this._container) {
-            width = this._container.getBoundingClientRect().width || 400;
-            height = this._container.getBoundingClientRect().height || 300;
+            const width = this._container.getBoundingClientRect().width || 400;
+            const height = this._container.getBoundingClientRect().height || 300;
+
+            const getTransformValues = (el) => {
+                while (el) {
+                    const transformMatrix = window.getComputedStyle(el).transform;
+                    if (transformMatrix && transformMatrix !== 'none') return transformMatrix.match(/matrix.*\((.+)\)/)[1].split(', ');
+                    if (!el.parentElement) return;
+                    el = el.parentElement;
+                }
+            };
+            const transformValues = getTransformValues(this._container);
```

Comment:
Nit: this could be inlined:

```suggestion
            let transformValues;
            let el = this._container;
            while (el && !transformValues) {
                const transformMatrix = window.getComputedStyle(el).transform;
                if (transformMatrix && transformMatrix !== 'none') transformValues = transformMatrix.match(/matrix.*\((.+)\)/)[1].split(', ');
                el = el.parentElement;
            }
```

Reply:
Added in your suggestion if you can give another look over! Thanks @mourner